app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,41 @@
+class ApplicationsController < ApplicationController
+  def show
+    @application = Application.find(params[:id])
+    @pets = @application.pets
+    if params[:search] == "pet" && params[:pet_name].present? 
+      pet_name = params[:pet_name]
+      @search_pets = Pet.where("name ILIKE ?", "%#{pet_name}%")
+    else 
+      @search_pets = []
```

Comment:
Instead of having an empty instance variable, it's slightly more conventional to have a check in the view, something like: `if @search_pets.present?`